src/views/clusteroverview/OverviewTab/metric-charts-card/utils/hooks/useTimespan.ts

```diff
@@ -0,0 +1,27 @@
+import { V1VirtualMachine } from '@kubevirt-ui/kubevirt-api/kubevirt';
+import DurationOption from '@kubevirt-utils/components/DurationOption/DurationOption';
+import { VirtualMachineModelGroupVersionKind } from '@kubevirt-utils/models';
+import { useK8sWatchResource } from '@openshift-console/dynamic-plugin-sdk';
+
+const useTimespan = (namespace?: string): number => {
+  const [vms] = useK8sWatchResource<V1VirtualMachine[]>({
+    groupVersionKind: VirtualMachineModelGroupVersionKind,
+    isList: true,
+    namespace,
```

Comment:
if `namespace` is `null`, here we'll fetch all vms in the cluster. 
Can this page be reached by non-priv user?

Reply:
Yes this page can be reached by non-admin, I know this logic isn't optimal, but deciding on 7 days as an arbitrary timespan will get buggy for namespaces that just created vms.
We need to discuss this issue with UX to think about redesigning this part to maybe not render to users without `list` permissions

Reply:
maybe we should add a time range input, and this would be the best solution IMO
@upalatucci @metalice @pcbailey @hstastna 
I would like your opinions as well on this subject.
/wip

Reply:
U can simply default it to a day or an hour.. Tell us the issue I'm sure they will agree

Reply:
That's also a valid option, will discuss this with them as well, thanks :) @metalice

Reply:
Discussed with Ronen, we keep the 7 days timespan and changing the messages a bit